x-pack/plugins/cases/public/components/user_actions/assignees.tsx

```diff
@@ -5,13 +5,166 @@
  * 2.0.
  */
 
-import type { UserActionBuilder } from './types';
+import { EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { UserProfileWithAvatar } from '@kbn/user-profile-components';
+import React, { memo } from 'react';
+import { SnakeToCamelCase } from '../../../common/types';
+import { Actions, AssigneesUserAction, User } from '../../../common/api';
+import { getName } from '../user_profiles/display_name';
+import { Assignee } from '../user_profiles/types';
+import { UserToolTip } from '../user_profiles/user_tooltip';
+import { createCommonUpdateUserActionBuilder } from './common';
+import type { UserActionBuilder, UserActionResponse } from './types';
+import * as i18n from './translations';
+import { getUsernameDataTestSubj } from '../user_profiles/data_test_subject';
+
+const FormatListItem: React.FC<{
+  children: React.ReactElement;
+  index: number;
+  listSize: number;
+}> = ({ children, index, listSize }) => {
+  if (shouldAddAnd(index, listSize)) {
+    return (
+      <>
+        {i18n.AND_SPACE}
+        {children}
+      </>
+    );
+  } else if (shouldAddComma(index, listSize)) {
+    return (
+      <>
+        {children}
+        {','}
+      </>
+    );
+  }
+
+  return children;
+};
+FormatListItem.displayName = 'FormatListItem';
+
+const shouldAddComma = (index: number, arrayLength: number) => {
+  return arrayLength > 0 && index !== arrayLength - 1;
+};
+
+const shouldAddAnd = (index: number, arrayLength: number) => {
+  return arrayLength > 1 && index === arrayLength - 1;
+};
+
+const Themselves: React.FC<{
+  index: number;
+  numOfAssigness: number;
+}> = ({ index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <>{i18n.THEMSELVES}</>
+  </FormatListItem>
+);
+Themselves.displayName = 'Themselves';
+
+const AssigneeComponent: React.FC<{
+  assignee: Assignee;
+  index: number;
+  numOfAssigness: number;
+}> = ({ assignee, index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <UserToolTip profile={assignee.profile}>
+      <strong>{getName(assignee.profile?.user)}</strong>
+    </UserToolTip>
+  </FormatListItem>
+);
+AssigneeComponent.displayName = 'Assignee';
+
+interface AssigneesProps {
+  assignees: Assignee[];
+  createdByUser: SnakeToCamelCase<User>;
+}
+
+const AssigneesComponent = ({ assignees, createdByUser }: AssigneesProps) => (
+  <>
+    {assignees.length > 0 && (
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
```

Comment:
For long lists of assignees, I think it looks better if you add `wrap` to the `EuiFlexGroup`.

Without:
<img width="688" alt="image" src="https://user-images.githubusercontent.com/7405507/186896431-38fc3105-d03f-4cbd-98a4-10849eaf6dab.png">

With `wrap`:

<img width="713" alt="image" src="https://user-images.githubusercontent.com/7405507/186896539-b2cbca1f-630e-4d31-9e1d-25078f524a0c.png">

I wondered if the whole comment `elastic assigned <user list> 9 days ago` could be made to wrap as a single line, but I think that would require more major refactoring so probably isn't worth the effort :)